### PR TITLE
Remove uses of Joda from xpack tests

### PR DESCRIPTION
**Pull request by rjernst**
Joda was deprecated internally many years ago, but there still exist
some tests that use it to format dates. This commit converts tests in
xpack that need date parsing or printing to use Java time.

### CODE REVIEW

**Comment by rjernst** [comment]
@elasticmachine update branch

File: x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/job/RollupIndexerIndexingTests.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public void testSimpleDateHisto() throws Exception {
                         "the_histo.date_histogram._count",
                         2,
                         "the_histo.date_histogram.time_zone",
-                        DateTimeZone.UTC.toString(),
+                        "UTC", // TODO: the default is hardcoded from Joda, we should change this
```
Comment (pgomulka) [inline x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/job/RollupIndexerIndexingTests.java:140]: maybe java.time.ZoneId.of("UTC").toString?

**Review by pgomulka — APPROVED** [review]
LGTM, left one comment. No need for another review

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Pull request merged**